### ISSUE TITLE
Notifications for PR review

### ISSUE THREAD

**Issue by AetherUnbound** [WordPress/openverse#158]
Given the pace of the project, it's easy to forget to review PRs. We've all noticed that PRs opened all week always get reviewed once they're called out in the next meeting, which isn't ideal because it slows down work and blocks progress.

The two-pronged approach to improving the system is
- tagging PRs with "priority" labels
  - each levels of priority is assigned a time period in which the review is expected
- notifying reviewers after a time period (decided based on the priority label)
  - preferably in the DMs or by tagging them in a comment on said PR

This approach ensures that our PRs get timely reviews. This discussion aims to reach a good agreement on the following aspects of the project.

- levels of priority and meaning of each level
- ETAs associated with each level
- number of notifications
- fraction of ETA at which to send notifications
- medium of notification
- implementation of the notifications
- worst case scenario / last resort

Details on these are listed below in specific comments. Please reply with your ideas, criticisms of submitted ideas and general up or down votes.

_Originally posted by @dhruvkb in https://github.com/WordPress/openverse/discussions/118_

**Comment by AetherUnbound** [comment]
Here is what has been discussed so far:

## Levels of priority and meaning at each level

Continuing with the existing priority levels, using the following definitions:

| Label | Meaning | Color |
|-|-|-|
| 🟥 priority: critical | Must be fixed ASAP | `#b60205` |
| 🟧 priority: high | Stalls work on the project or its dependents | `#ff9f1c` |
| 🟨 priority: medium | Not blocking but should be fixed soon | `#ffcc00` |
| 🟩 priority: low | Low priority and doesn't need to be rushed | `#cfda2c` |

## ETAs associated with each level

This appeared to be our majority consensus, although we this could be discussed further:


| Label | ETA/Review urgency |
| -- | -- |
| critical | 1d |
| high | 1d (but review it _after_ critical PRs and if you run out of time put this at the top of your list for the next day) |
| medium | 5d (but try to review it as soon as you can and definitely before low priority PRs) |
| low | 5d (but review these last unless they are at the fifth day, in which case delay other work to review any PRs that have reached threshold including medium and low) |

## Number of notifications

Team consensus was 1 notification.

## Fraction of ETA at which to send notifications

0.75 of the above interval depending on priority was generally accepted.

## Medium of notification

It appears that general team opinion was:
1. Start with GitHub pings (potentially a GitHub post @-ing folks)
2. Move to Slack if we feel that would be better, but it should be the Make WP Slack.

## Implementation

Still undecided, though likely a daily/every 12h scheduled GitHub action would work to start.

## Last resort

Folks are OK with pinging individuals or the team as a whole!

## Other thoughts

GitHub has some mechanism for scheduled reminders:
* https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-your-membership-in-organizations/managing-your-scheduled-reminders
* https://docs.github.com/en/organizations/organizing-members-into-teams/managing-scheduled-reminders-for-your-team
* https://github.com/settings/reminders

**Comment by zackkrida** [comment]
@WordPress/openverse-developers Revisiting this, how are we feeling? Some example reminders currently post to #openverse-firehose in the Making WP Slack, and they look like this:

<img width="602" alt="CleanShot 2022-03-23 at 18 06 09@2x" src="https://user-images.githubusercontent.com/6351754/159803910-159ec7ee-d4d3-4507-b78a-15bb22f3711a.png">

<details>
<summary>Text Version:</summary>

Pending review on WordPress/openverse - Manage reminder
[#186] Add an RFC for working with feature flags (dhruvkb)
2 days stale · 13 days old · Waiting on AetherUnbound, stacimc
[#193] RFC: Monitoring across the stack (sarayourfriend)
a minute stale · 6 days old · Waiting on rbadillap, dhruvkb, stacimc
[#194] Add infrastructure repo to synced repo list (AetherUnbound)
16 hours stale · 16 hours old · Waiting on sarayourfriend
</details>

I would recommend looking at the [configuration settings](https://github.com/orgs/WordPress/teams/openverse/settings/reminders/40911) for these reminders too if anyone hasn't.

**Comment by AetherUnbound** [comment]
I'm loving the github reminders in #openverse-firehose! I check it every day :)

**Comment by sarayourfriend** [comment]
We implemented this as a DAG: https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/maintenance/pr_review_reminders/pr_review_reminders.py
